File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    var answer = 0
    let losts = lost.filter(i => !reserve.includes(i));
    let reserves = reserve.filter(i => !lost.includes(i));

    reserves.sort((a, b) => a - b);

    for (let i of reserves) {
        if (losts.includes(i-1)) {
            losts = losts.filter(x => x !== i - 1);
        } else if (losts.includes(i+1)) {
            losts = losts.filter(x => x !== i + 1);
        }
    }

    answer = n-losts.length;
    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    losts = set(lost)-set(reserve)
    reserves = set(reserve)-set(lost)

    for i in sorted(reserves):
        if i-1 in losts:
            losts.remove(i-1)
        elif i+1 in losts:
            losts.remove(i+1)

    answer = n-len(losts)
    return answer
